subroutines.sh

```diff
@@ -13,35 +13,110 @@ send_command() {
+         break
+        else
      log_error "tesla-control send command failed exit status $EXIT_STATUS."
-	 log_info $message
-	 log_notice "Retrying in $SEND_CMD_RETRY_DELAY seconds"
-	fi
+         log_info $message
+         log_notice "Retrying in $SEND_CMD_RETRY_DELAY seconds"
+        fi
     sleep $SEND_CMD_RETRY_DELAY
   fi
  done
 }
 
+# Tesla VIN to BLE Local Name
+tesla_vin2ble_ln() {
+  TESLA_VIN=$1
+  BLE_LN=""
+
+  log_debug "Calculating BLE Local Name for Tesla VIN $TESLA_VIN"
+  VIN_HASH="$(echo -n ${TESLA_VIN} | sha1sum)"
+  # BLE Local Name
+  BLE_LN="S${VIN_HASH:0:16}C"
+  log_debug "BLE Local Name for Tesla VIN $TESLA_VIN is $BLE_LN"
+
+  echo $BLE_LN
+
+}
+
 listen_to_ble() {
- log_notice "Listening to BLE for presence"
- log_warning "Needs updating for multi-car, only supports TESLA_VIN1 at this time. Doesn't support deprecated TESLA_VIN usage"
- PRESENCE_TIMEOUT=5
- set +e
- bluetoothctl --timeout $PRESENCE_TIMEOUT scan on | grep $BLE_MAC1
- EXIT_STATUS=$?
- set -e
- if [ $EXIT_STATUS -eq 0 ]; then
-   echo "$BLE_MAC1 presence detected"
-   eval $MOSQUITTO_SUB --nodelay -t tesla_ble_mqtt/$TESLA_VIN1/binary_sensor/presence -m ON
- else
-   echo "$BLE_MAC1 presence not detected"
-   eval $MOSQUITTO_SUB --nodelay -t tesla_ble_mqtt/$TESLA_VIN1/binary_sensor/presence -m OFF
- fi
+  n_cars={$1:-3}
+
+  log_notice "Listening to BLE for presence"
+  log_warning "Doesn't support to deprecate previous TESLA_VIN usage"
+  PRESENCE_TIMEOUT=10
+  set +e
+  BLTCTL_OUT=$(bluetoothctl --timeout $PRESENCE_TIMEOUT scan on | grep $BLE_MAC1 2>&1)
+  set -e
+  log_debug "${BLTCTL_OUT}"
+  for count in $(seq $n_cars); do
+    BLE_LN=$(eval echo "echo \$BLE_LN${count}")
+    BLE_MAC=$(eval "echo \$BLE_MAC${count}")
+    PRESENCE_EXPIRE_TIME=$(eval "echo \$PRESENCE_EXPIRE_TIME${count}")
+    TESLA_VIN=$(eval "echo \$TESLA_VIN${count}")
+
+    MQTT_TOPIC="tesla_ble_mqtt/$TESLA_VIN/binary_sensor/presence"
+
+    if echo "$(BLTCTL_OUT)" | grep -q $BLE_MAC; then
+      log_info "BLE MAC $BLE_MAC presence detected"
+      EPOCH_TIME=$(date +%s)
+      # We need a function for mosquitto_pub w/ retry
+      if [ $EPOCH_TIMW < $PRESENCE_EXPIRE_TIME ]; then
+        log_info "TESLA VIN $TESLA_VIN ($BLE_MAC) TTL expired, update mqtt topic with presence ON"
+        set +e
+        MQTT_OUT=$(eval $MOSQUITTO_SUB --nodelay -t "$MQTT_TOPIC" -m ON 2>&1)
+        EXIT_CODE=$?
+        set -e
+        [ $EXIT_CODE -ne 0 ] \
+          && log_error "$(MQTT_OUT)" \
+          && continue
+        log_info "mqtt topic "$MQTT_TOPIC" succesfully updated to ON"
+
+        # Updating Presence Expire Time in Epoch
+        EPOCH_TIMW=$(date +%s)
+        EPOCH_EXPIRE_TIME=$(expr EPOCH_TIME + $BLE_PRESENCE_TTL)
+        PRESENCE_EXPIRE_TIME${count}=$EPOCH_EXPIRE_TIME
+        log_debug "VIN $TESLA_VIN $BLE_MAC update Presence Expire Time to $EPOCH_EXPIRE_TIME"
+      else
+        log_info "VIN $TESLA_VIN ($BLE_MAC) TTL has not expires at $PRESENCE_EXPIRE_TIME"
```

Comment:
Typo? expires --> expired

Reply:
Thank you again for going over this PR.

Fixed - I had noticed that one. Bad English at ~ 4 o'clock :)  I even remember when I wrote it I hesitated and couldn't even figure out if what I had picked was right or not 🤣